### PR TITLE
feat(backend): add configurable S3 path style support

### PR DESCRIPTION
**Pull request by ntny**
**Description of your changes:**
Resolve: #10814

This MR introduces the ability to configure the S3 path style 
Currently it is hardcoded [here](https://github.com/kubeflow/pipelines/blob/master/backend/src/v2/objectstore/object_store.go#L261) as pathStyle
'PathStyle' will be the default setting

**Checklist:**
- [ ] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/src/v2/config/s3.go [hunk 1]
```diff
@@ -33,7 +33,8 @@ type S3ProviderDefault struct {
 	// optional for any non aws s3 provider
 	Region string `json:"region"`
 	// optional
-	DisableSSL *bool `json:"disableSSL"`
+	DisableSSL     *bool `json:"disableSSL"`
+	ForcePathStyle *bool `json:"forcePathStyle"`
```
Comment (hbelmiro) [inline backend/src/v2/config/s3.go:37]: nit: This is optional, right? May be worth a comment, like in the above fields.

File: backend/src/v2/config/s3.go [hunk 2]
```diff
@@ -51,7 +52,8 @@ type S3Override struct {
 	BucketName string `json:"bucketName"`
 	KeyPrefix  string `json:"keyPrefix"`
 	// required
-	Credentials *S3Credentials `json:"credentials"`
+	Credentials    *S3Credentials `json:"credentials"`
+	ForcePathStyle *bool          `json:"forcePathStyle"`
```
Comment (hbelmiro) [inline backend/src/v2/config/s3.go:56]: Same as above.

**Comment by HumairAK** [comment]
/lgtm
/approve 

Thanks @ntny ! 
Can you also update the documentation here: https://github.com/kubeflow/website/blob/master/content/en/docs/components/pipelines/operator-guides/configure-object-store.md

**Pull request merged**